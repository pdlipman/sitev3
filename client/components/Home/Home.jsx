import React from 'react';

import IconHeader from '../Layout/Header/IconHeader.jsx';

const Home = () => (
    <div>
        <IconHeader
            contents='Home'
            headerTag='h2'
            iconName='home'
            subHeader={'Welcome to my site!'}
        />
        This is the home page.
    </div>
);

export default Home;

