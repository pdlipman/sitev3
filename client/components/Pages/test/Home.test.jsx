import React from 'react';
import { shallow } from 'enzyme';

import Home from '../Home.jsx';

describe('Pages/Home.jsx', () => {
    it('Home should render properly', () => {
        const wrapper = shallow(
            <Home>Hello Jest!</Home>,
        );
        expect(wrapper).toMatchSnapshot();
    });

});
