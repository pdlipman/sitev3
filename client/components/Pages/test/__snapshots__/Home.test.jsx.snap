// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Home.jsx Home should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home>
    Hello Jest!
  </Home>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <IconHeader
          contents="Home"
          headerTag="h2"
          iconName="home"
          subHeader="Welcome to my site!"
        />,
        "This is the home page.",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "contents": "Home",
          "headerTag": "h2",
          "iconName": "home",
          "subHeader": "Welcome to my site!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "This is the home page.",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <IconHeader
            contents="Home"
            headerTag="h2"
            iconName="home"
            subHeader="Welcome to my site!"
          />,
          "This is the home page.",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "contents": "Home",
            "headerTag": "h2",
            "iconName": "home",
            "subHeader": "Welcome to my site!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "This is the home page.",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
